pkg/endpoint/bpf.go

```diff
@@ -886,6 +886,15 @@ func (e *Endpoint) addPolicyKey(keyToAdd policy.Key, entry policy.MapStateEntry,
 	return true
 }
 
+// ApplyPolicyMapChanges updates the Endpoint's PolicyMap with the changes
+// that have accumulated for the PolicyMap via various outside events (e.g.,
+// identities added / deleted).
+func (e *Endpoint) ApplyPolicyMapChanges() error {
+	e.mutex.Lock()
```

Comment:
As discussed offline we should make this a `lockAlive` - not sure if we will want to log the error in that case in the caller; perhaps we can switch on the error type and only log if the error is not a liveness error.

Reply:
`UpdatePolicyMaps` warns on error. Did you mean that we would switch in there or in `ApplyPolicyMapChanges`?

Reply:
I'm ambivalent - I guess it's better to return an error and let the caller do what if wants with it?